### PR TITLE
Run expects separately per module, report test results per module behind `--verbose` flag

### LINKED ISSUE #6490: `roc test platform/main.roc` should test every file referenced in exposes and the respective imports of those files too
`roc test platform/main.roc` currently only tests what's imported.

Note that these files begin with the platform keyword like:
```
platform "webserver"
requires {} { main : Request -> Task Response [] }
    exposes [
...
```

### LINKED ISSUE #6491: `roc test` should print filenames of tested files when recursively testing imports
Instead of the current:
```
0 failed and 2 passed in 129 ms.
```
This should be something like:
```
main.roc:
    0 failed and 1 passed in 52 ms.
someImportOfMain.roc:
    0 failed and 1 passed in 46 ms.
```

### PR DESCRIPTION
**Pull request by jschear**
Closes #6490, closes #6491.

#6490 seemed to already be working.

Here's some example output (with some intentionally failing tests):

```
❯ cargo run -- test ../roc-parser/package/main.roc --verbose
   Compiling roc_cli v0.0.1 (/Users/jschear/dev/roc/roc/crates/cli)
    Finished dev [unoptimized + debuginfo] target(s) in 10.59s
     Running `target/debug/roc test ../roc-parser/package/main.roc --verbose`
── EXPECT FAILED in ../roc-parser/package/String.roc ───────────────────────────

This expectation failed:

289│  expect parseStr parseNumbers "1,2,3" == Ok [1, 2, 4]
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^


── EXPECT FAILED in ../roc-parser/package/HTTP.roc ─────────────────────────────

This expectation failed:

73│>  expect
74│>      actual = parseStr httpVersion "HTTP/1.2"
75│>      expected = Ok { major: 1, minor: 1 }
76│>      actual == expected

When it failed, these variables had these values:

actual : Result {
    major : U8,
    minor : U8,
} [
    ParsingFailure Str,
    ParsingIncomplete Str,
]
actual = Ok { major: 1, minor: 2 }

expected : [
    Err [
        ParsingFailure Str,
        ParsingIncomplete Str,
    ],
    Ok {
        major : U8,
        minor : U8,
    },
]
expected = Ok { major: 1, minor: 1 }


Compiled in 1591 ms.

Core.roc:
    0 failed and 3 passed in 0 ms.

String.roc:
    1 failed and 22 passed in 0 ms.

HTTP.roc:
    1 failed and 6 passed in 2 ms.

```

Without the `--verbose` flag:

```
❯ cargo run -- test ../roc-parser/package/main.roc
    Finished dev [unoptimized + debuginfo] target(s) in 0.51s

... same as above ...

2 failed and 31 passed in 1696 ms.
```

Discussed in https://roc.zulipchat.com/#narrow/stream/316715-contributing/topic/Test.20discovery

### CODE REVIEW

File: crates/cli/src/lib.rs [hunk 1]
```diff
@@ -567,22 +580,50 @@ pub fn test(matches: &ArgMatches, triple: Triple) -> io::Result<i32> {
...
+        ANSI_STYLE_CODES.red
+    };
 
+    let passed_color = ANSI_STYLE_CODES.green;
+    let reset = ANSI_STYLE_CODES.reset;
+
+    if show_module_label {
+        let module_name = module_id.to_ident_str(interns);
+        println!(
+            "\n{module_name}.roc:\n    {failed_color}{failed_count}{reset} failed and {passed_color}{passed_count}{reset} passed in {} ms.\n",
```
Comment (jschear) [inline crates/cli/src/lib.rs:619]: Is there a more proper way to get the name of the source file here?

Comment (miilyn) [reply crates/cli/src/lib.rs:619]: One (possibly dirty) way to get the module's file path is to make use of `expectations`, 
which is a `VecMap<ModuleId, Expectations>`.

Given a `module_id` you can then get to its `Expectations` and thus to the `module`'s path via:
```roc
expectations.get(&module_id).unwrap().path
```

File: crates/cli/tests/expects_transitive/Direct.roc [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    imports [
+        Transitive,
+    ]
+
+addAndStringify = \num1, num2 ->
+    Num.toStr (Transitive.add num1 num2)
+
+expect addAndStringify 1 2 == "3"
+
+expect addAndStringify 3 4 == "7"
```
Comment (miilyn) [inline crates/cli/tests/expects_transitive/Direct.roc:14]: Someone more experienced should give advice how relevant it is to test for both `expect` and `expect-fx`.

When in doubt, personally I'd like to test for both -- just to be safe.

File: crates/cli/src/lib.rs (hunk 1 above)
Comment (Anton-4) [reply crates/cli/src/lib.rs:619]: We should probably pull out shared code from `report.rs:pretty_header_with_path` but we can do that in a follow-up PR.

File: crates/cli/src/lib.rs [hunk 3]
```diff
@@ -542,21 +544,32 @@ pub fn test(matches: &ArgMatches, triple: Triple) -> io::Result<i32> {
...
-        expects,
-    )
-    .unwrap();
+    let mut total_failed_count = 0;
+    let mut total_passed_count = 0;
+
+    let mut results_by_module = Vec::new();
+    let global_layout_interner = layout_interner.into_global();
+    for (module_id, expects) in expects_by_module.into_iter() {
+        let test_start_time = Instant::now();
```
Comment (jschear) [inline crates/cli/src/lib.rs:553]: One thing to note here: previously the test duration we reported included the time taken by `load_and_monomorphize`, `expect_mono_module_to_dylib`, etc.; now the per-module time only includes the time it took to actually run the tests. I think that's the only way to sensibly split up the duration, but it means that we often report a value of 0ms: `1 failed and 22 passed in 0 ms.`

**Comment by Anton-4** [comment]
This looks pretty good so far @jschear :)

I made some minor stylistic changes, I'll finish up my review on Monday.

**Comment by jschear** [comment]
Sounds good -- appreciate the review & your changes look good to me. Much more readable.

**Comment by Anton-4** [comment]
Today was a busy day, I should be able to finish this up tomorrow :crossed_fingers:

File: crates/cli/tests/expects_transitive/Direct.roc (hunk 2 above)
Comment (Anton-4) [reply crates/cli/tests/expects_transitive/Direct.roc:14]: `expect-fx` is not completely implemented yet, so I think we should leave it out for now.

File: crates/cli/src/lib.rs (hunk 3 above)
Comment (Anton-4) [reply crates/cli/src/lib.rs:561]: I've asked for some extra input [on zulip](https://roc.zulipchat.com/#narrow/stream/395097-compiler-development/topic/test.20time.20reporting/near/426187328).

**Comment by Anton-4** [comment]
I can implement [your proposal](https://roc.zulipchat.com/#narrow/stream/395097-compiler-development/topic/test.20time.20reporting/near/426395448) on Friday, or if you can get to it first that's fine for me as well :)

**Comment by jschear** [comment]
@Anton-4 I'll take a stab at it today 👍

**Review by Anton-4 — DISMISSED** [review]
Thanks @jschear!

**Review by Anton-4 — APPROVED** [review]

**Pull request merged**